### DISCUSSION TITLE
wizard and submit

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#5828] [Help] [answered]
In my component I use form with wizard.

In my blade I have 

 ```
<form wire:submit.prevent="submit">
            {{ $this->form }}
            <div class="mt-5">
                <button type="submit">
                    Send 
                </button>
            </div>
   </form>
```
You can only reach next step, if the actual is valid. Fine!
BUT: How can I show the submit-button only, if the last wizards step is shown?

**Comment by danharrin** [comment, marked as answer]
Please read the docs https://filamentphp.com/docs/2.x/forms/layout#wizard

**Reply by photojo** [reply]
Thank you!

**Reply by intrepidws** [reply]
@danharrin I'm doing this but it seems that validation isn't performed when doing this? is that expected behavior or am I screwing something up?

**Reply by danharrin** [reply]
You're screwing something up 😂

**Reply by intrepidws** [reply]
Just to be clear, validation occurs on every step _except_ the final step, where the submit button exists.

**Reply by danharrin** [reply]
What is in your submit function? You need `$this->form->getState()` there like any other form. We handle intra-step validation but not on submit, because at that point the entire form should be validated again, not just a specific step.

**Reply by intrepidws** [reply]
Blarg. My final step had only checkbox fields and the "required" method simply confirms that the field is present, so the validation wasn't triggering anything when I didn't check it. I had to change to using the "accepted" rule. Thanks for your help.
